source/framework/core/src/TRestRun.cxx

```diff
@@ -1021,17 +1011,19 @@ TFile* TRestRun::MergeToOutputFile(vector<string> filenames, string outputfilena
 }
 
 ///////////////////////////////////////////////
-/// \brief Create a new TFile as REST output file. Writing metadata objects into
-/// it.
+/// \brief Create a new TFile as REST output file. Writing metadata objects into it.
 ///
 TFile* TRestRun::FormOutputFile() {
     CloseFile();
+
     fOutputFileName = FormFormat(fOutputFileName);
+    // remove unwanted "./" etc. from the path while resolving them
+    fOutputFileName = std::filesystem::weakly_canonical(fOutputFileName.Data());
+
     fOutputFile = new TFile(fOutputFileName, "recreate");
     fAnalysisTree = new TRestAnalysisTree("AnalysisTree", "AnalysisTree");
     fEventTree = new TTree("EventTree", "EventTree");
-    // fAnalysisTree->CreateBranches();
```

Comment:
Why this is not needed anymore? Perhaps @nkx has some comments here?

Reply:
> Why this is not needed anymore? Perhaps @nkx has some comments here?

No idea, I will @nkx111 reply.